src/v2i-hub/TimPlugin/src/TimPlugin.cpp

```diff
@@ -0,0 +1,438 @@
+#include "TimPlugin.h"
+#include <WGS84Point.h>
+#include "TimeHelper.h"
+#include "XmlCurveParser.h"
+#include "VehicleLocate.h"
+#include <tmx/messages/IvpDmsControlMsg.h>
+#include <tmx/j2735_messages/TravelerInformationMessage.hpp>
+
+using namespace std;
+using namespace tmx::messages;
+using namespace tmx::utils;
+
+namespace TimPlugin {
+
+
+
+
+/**
+ * Construct a new EpcwPlugin with the given name.
+ *
+ * @param name The name to give the plugin for identification purposes
+ */
+TimPlugin::TimPlugin(string name) :
+		PluginClient(name) {
+
+	AddMessageFilter < BsmMessage > (this, &TimPlugin::HandleBasicSafetyMessage);
+
+	SubscribeToMessages();
+
+}
+
+TimPlugin::~TimPlugin() {
+}
+
+void TimPlugin::UpdateConfigSettings() {
+
+	GetConfigValue<uint64_t>("Frequency", _frequency);
+
+	{
+		lock_guard<mutex> lock(_mapFileLock);
+		if (GetConfigValue<string>("MapFile", _mapFile))
+			_isMapFileNew = true;
+	}
+
+	GetConfigValue<uint64_t>("Snap Interval", _snapInterval);
+	GetConfigValue<uint64_t>("Vehicle Timeout", _vehicleTimeout);
+	GetConfigValue<string>("Start Broadcast Date (mm-dd-YYYY):", _startTimDate);
+	GetConfigValue<string>("Stop Broadcast Date (mm-dd-YYYY):", _stopTimDate);
+	GetConfigValue<string>("Start Broadcast Time (HH:MM:SS):", _startTimTime);
+	GetConfigValue<string>("Stop Broadcast Time (HH:MM:SS):", _stopTimTime);
+
+}
+
+void TimPlugin::OnConfigChanged(const char *key, const char *value) {
+	PluginClient::OnConfigChanged(key, value);
+	UpdateConfigSettings();
+}
+
+
+void TimPlugin::HandleBasicSafetyMessage(BsmMessage &msg,
+		routeable_message &routeableMsg) {
+
+PLOG(logDEBUG)<<"HandleBasicSafetyMessage";
```

Comment:
Is this needed?  Aren't we broadcasting TIM irrespective of where or if there is vehicle?

Reply:
Addressed